lua/rocks-config/init.lua

```diff
@@ -65,18 +77,30 @@ local function try_load_config(mod_name)
         return true
     end
 
-    -- Loaders that search `package.preload` and `package.path`.
-    -- We don't need to search the cpath or neovim's runtimepath,
-    -- as the nvim `lua` directory is added to the `package.path`.
-    for _, searcher in ipairs(package.loaders) do
-        local loaded = try_load_like_require(searcher, mod_name)
+    local loader = try_get_loader_for_module(mod_name)
 
-        if loaded ~= nil then
-            package.loaded[mod_name] = loaded
-            return true
-        end
+    if loader == nil then
+        return false
+    end
+
+    package.loaded[mod_name] = load_like_require(loader)
+
+    return true
+end
+
+---Checks if a plugin that already had a configuration loaded has
+---a given duplicate candidate configuration, and warns the user.
+---@param plugin_name string The plugin that is being configured
+---@param mod_name string The configuration module name to check for
```

Comment:
issue: missing `@param`
readability: I would suggest a more meaningful name than `match_name`

```suggestion
---@param plugin_name string The plugin that is being configured
---@param base_filename string The configuration's base file name
---@param mod_name string The configuration module name to check for
```

Reply:
I changed it to `config_basename` and added the `@param`. What do you think?

Reply:
Looks good :smile: